### ISSUE TITLE
% regg foods (BB #159)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#159]
This issue was migrated from bitbucket.
**Original Reporter:** niyo
**Original Date:** 27.05.2010 16:01:36 GMT+0000
**Original Priority:** major
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/159

<hr>

You can stack the procentual regeneration food with every other food and drinks

for example you can eat 2 % regg foods + 1 drink + 1 eat

These are the items:

http://www.wowhead.com/item=21215
http://www.wowhead.com/item=21537

see http://www.oregoncore.com/index.php?/topic/309-regg-foods/

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 13.01.2014 19:32:24 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/159/#comment-7714062

<hr>

Can't reproduce with those two items on my core currently. Requires testing by Yooki and then we can confirm that this issue is resolved.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Yooki
**Original Date:** 13.01.2014 19:46:17 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/159/#comment-7714265

<hr>

Confirmed! Food and drink consumable effects indeed do stack.
I am surprised this was overlooked by Oregon to begin with.

We should really start assigning people to issues, just to keep track of what issues are currently being resolved by a developer.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 20.01.2014 17:29:46 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/159/#comment-7829235

<hr>

Resolved issue #159 - Food & Drinks:

Food regeneration effects were permitted to stack, and this did not happen every time,
which reveals more inconsistencies in the core. Interesting enough this issue did not affect drinks
that regenerate a fixed amount of power. Note that I did not test the ones that regenerate percentages.

Okay, so the complete fix for this issue is not brought by this commit alone.
One of the latest commits (Player stats) restructured some code to disable
the core from adding/stacking auras and applying their modifiers if the
aura effects had the same ID. This prevented drinks and food that had the same effect from stacking.

As can be seen in the change-log, I've just added a check for what I believe are types of
auras that heal by percentage (SPELL_AURA_OBS_MOD_HEALTH). Because this was missing it caused
items such as Graccu's Mince Meat Fruitcake and Festival Dumplings's effects to stack together.

→ <<cset 288eab370449>>
